### DISCUSSION TITLE
Suggestions on the best approach to implement change of email address when users connect with Linkedin Oauth

### DISCUSSION THREAD

**Discussion by superseby2** [appwrite/appwrite#3573] [General]
Hi all, 

Our users connect with LinkedIn Oauth only (there is no username/password). We are launching the service and thought it would be better to start this way (as it is super simple to do with Appwrite)

Turns out, email addresses on LinkedIN profiles are very often outdated and we would like to be able to propose to our users to change theirs.

BUT : 
- if we manage to update their email using Appwrite's API (not even sure this is possible as we need to have a password which we have not when logging from Oauth) -> they won't be able to login at all because the match on the email address registered @ linkedin won't be there anymore
- If we create something like a "primary address" pointing to an alternate address, it would work : they will still be able to connect using LinkedIn AND we will be able to communicate with them using the primary address they defined 
BUT 
   -> we will have to consider this address as the "real one"  (and not the one from their account) everywhere (especially in functions) which could be confusing
   -> when we decide to add other authentication methods like user/password, we won't be able to take the benefit of using Appwrite's built-in api related to email management which is a bummer. (or we could but that would mean more confusion as we would handle users differently depending on the authentication method)

The alternate primary email address solution seems to me the only viable still ... (another option is simply ditch LinkedIn Oauth and go fully email/password of course .. but even with Appwrite this would take some time to implement)

Any suggestions on the best way to handle this use case is more than welcome :sweat_smile: 

Thank you!

**Comment by stnguyen90** [comment]
It sounds like you would like to be able to have multiple emails for the same user. Is that correct?

As of now, since each user only has 1 email, you would have to use a separate collection to maintain the "primary" email address. Some apps refer to it as a "communication" or "contact" email address.
